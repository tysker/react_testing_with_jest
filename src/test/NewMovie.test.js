import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import NewMovie from '../components/NewMovie';

afterEach(cleanup);

test( '<NewMovie>', () => {
    const { debug, getByTestId, queryByTestId, container, getByText } = render(<NewMovie/>);
    const pagetitle = getByTestId('page-title');

    expect(pagetitle.textContent).toBe('New Movie');
    expect(queryByTestId('movie-form')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
})