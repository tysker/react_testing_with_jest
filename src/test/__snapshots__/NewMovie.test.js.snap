// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie> 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    New Movie
  </h1>
  <form
    data-testid="movie-form"
  >
    <label
      for="text"
    >
      Text
      <input
        id="text"
        type="text"
      />
    </label>
    <button>
      Submit
    </button>
  </form>
</div>
`;
